### PR TITLE
CI: PyPy3 testing

### LINKED ISSUE #37653: ENH: PyPy3 testing
#### Is your feature request related to a problem?

With the increasing support for PyPy3 in NumPy, it would be helpful if pandas started testing against PyPy3.

#### Describe the solution you'd like

Please consider adding `pypy3` to the `.travis.yml` (allowed to fail for now).

#### API breaking implications

No expected API breaks

#### Describe alternatives you've considered

This is more of a portability concern than anything else.

#### Additional context

Something like:

```yaml
  matrix:
   -  python: pypy3
      env:
        - JOB="pypy3" PATTERN="(not slow and not network and not clipboard)"
  allow_failures:
   -  python: pypy3
      env:
        - JOB="pypy3" PATTERN="(not slow and not network and not clipboard)"
```

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] xref #37653 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by lithomas1** [comment]
Wow, only 1 failing test here. I guess the crashes/segfaults are probably due to an older numpy then(this needs to be verified though).

File: ci/deps/actions-pypy-38.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  # tools
+  - cython>=0.29.24
+  - pytest>=6.0
+  - pytest-cov
+  - pytest-xdist>=1.31
+  - hypothesis>=5.5.3
+
+  # required
+  - numpy<1.20  # GH#39541 compat for pyarrow<3
```
Comment (jbrockmendel) [inline ci/deps/actions-pypy-38.yaml:18]: id rather have both newer pyarrow and numpy, why are you doing it the other way around?

Comment (lithomas1) [reply ci/deps/actions-pypy-38.yaml]: Thanks for the catch. I accidentally forgot to change this when I copy-pasted from the database builds.

**Comment by jreback** [comment]
thanks @lithomas1

**Pull request merged**